core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -174,7 +176,13 @@ public void commit(TableMetadata base, TableMetadata metadata) {
   @Override
   public FileIO io() {
     if (defaultFileIo == null) {
-      defaultFileIo = new HadoopFileIO(conf);
+      defaultFileIo = ClassLoaderUtil.fromProperty(
+          current() == null ? new HashMap<>() : current().properties(),
+          TableProperties.WRITE_FILE_IO_IMPL,
+          HadoopFileIO.class.getName(),
+          HadoopFileIO.class,
```

Comment:
The interface should be FileIO instead of HadoopFileIO.

Reply:
It's HadoopFileIO in the original class. Is it intended or a bug?

https://github.com/apache/iceberg/blob/master/core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java#L65

Reply:
Since the same concrete class was always instantiated, I think we used the class in case we needed anything specific to `HadoopFileIO`. But it can be `FileIO` since nothing specific is used.